eslint.config.cjs

```diff
@@ -1,50 +1,194 @@
+//     plugin	 -> A re-export of @typescript-eslint/eslint-plugin
+// Notes: We still need to install the parser and plugin separately
+// "@typescript-eslint/parser" –> This parser allows ESLint to understand TypeScript
+//    syntax. Without it, ESLint won’t be able to process TypeScript files correctly.
+// "@typescript-eslint/eslint-plugin" –> This ESLint plugin provides additional
+//    TypeScript-specific rules.
+//
+// Optional package used that are very useful for best practice:
+// "eslint-plugin-unicorn" –> A plugin that provides a collection of additional
+//   rules that enforce best practices and modern JavaScript standards. It’s
+//   optional but useful for enforcing stricter code quality.
+//
+// "eslint-formatter-stylish" –> A package that provides a stylish formatter for
+//    ESLint. It’s optional but useful for a more readable output.
+//-----------------------------------------------------------------------------
+
+//-----------------------------------------------------------------------------
+// May need to use the chai plugin in the future
+// npm install --save-dev eslint-plugin-chai-friendly
+// And configure it into plugins: { "chai-friendly": chaiPLugin }
+//-----------------------------------------------------------------------------
+
+//-----------------------------------------------------------------------------
+// Eslint rules Settings
+// Severity Levels ("off", "warn", "error")
+//   "off"   -> Disables the rule.
+//   "warn"  -> Show a warning, does not fail ESLint (does not affect exit code).
+//   "error" -> Shows an error and makes ESLint fail (exit code is 1 when triggered).
+// Examples:
+//   "no-console": "warn" -> Shows a warning when `console.log` is used
+//   "eqeqeq": "error"    -> Throws an error if `==` instead of `===` is used
+//   "semi": "off"        -> Turns off semicolon checking
+//
+// Rule-Specific Settings ("always", "never", and More) take additional configuration
+//   "always"     -> Enforces something (e.g., always require semicolons, newlines, etc.).
+//   "never"      -> Prevents something (e.g., never allow semicolons, newlines, etc.).
+//   "consistent" -> Enforces consistency (e.g., either all single-line or all multi-line).
+//   { "minItems": X } ->	Requires a new line if an array/object has X or more items.
+// Examples:
+//   "semi": ["error", "always"], // Requires semicolons
+//   "quotes": ["error", "never"], // Disallows quotes (uses backticks when possible)
+//   "array-bracket-newline": ["error", { "minItems": 3 }] Forces newline for arrays with 3+ items
+//-----------------------------------------------------------------------------
+const eslint = require('@eslint/js')
+const nodePlugin = require("eslint-plugin-n")
+const stylistic = require('@stylistic/eslint-plugin')
+const tsPlugin = require('@typescript-eslint/eslint-plugin')
+const tsParser = require('@typescript-eslint/parser')
+// const chaiPlugin = require("eslint-plugin-chai-friendly")
```

Comment:
What are your thoughts on leaving this out for now?

Reply:
I was just too tire to add it.